### DISCUSSION TITLE
why jax not tensorflow,pytorch

### DISCUSSION THREAD

**Discussion by quantum2689** [xai-org/grok-1#341] [Q&A]
why jax not tensorflow,pytorch

**Comment by anddea** [comment]
There are reasons to use one over the other and vice versa. I don't think anyone from XAI will answer your question, so your best choice would be asking an LLM 'why JAX over PyTorch and TF'

JAX is used for Gemma as well:
https://github.com/google-deepmind/gemma

Although they have an official implementation in PyTorch:
https://github.com/google/gemma_pytorch

**Comment by Tunar787** [comment]
Sex

**Comment by Tunar787** [comment]
Sex

**Comment by xxh031203** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
